drivers/overlay/encryption.go

```diff
@@ -43,42 +45,35 @@ func (k *key) String() string {
 	return ""
 }
 
-type spi struct {
-	forward int
-	reverse int
+// encrElem contains the encryption info needed to
+// program the ipsec tunnel between a pair of nodes.
+type encrElem struct {
+	forwardSPI int
+	reverseSPI int
+	forwardKey []byte
+	reverseKey []byte
 }
 
-func (s *spi) String() string {
-	return fmt.Sprintf("SPI(FWD: 0x%x, REV: 0x%x)", uint32(s.forward), uint32(s.reverse))
+func newEncrElem(src, dst net.IP, k *key) *encrElem {
+	return &encrElem{
+		forwardSPI: buildSPI(src, dst, k.tag),
+		reverseSPI: buildSPI(src, dst, k.tag),
+		forwardKey: buildKey(src, dst, k.tag, k.value),
```

Comment:
You're calling `buildKey(src, dst, k.tag, k.value)` on both `forwardKey` and `reverseKey`. Doesn't this mean both keys will be the same?